### ISSUE TITLE
pyright raises an error when accessing a private member of a class defined in a different file

### ISSUE THREAD

**Issue by timhughes** [microsoft/pyright#5825]
**Describe the bug**
pyright raises an error when accessing a private member of a class defined in a different file but is fine with a class defined in the same file. In the following I would expect pyright not to raise an error for `_last_mtime` as it doesn't give an error for  `_a` when it is used in class `_Bar`

**Code or Screenshots**
`example.py`
```python
from gzip import _GzipReader


class _Foo:
    def __init__(self):
        self._a = "a"

    def x_member(self):
        pass


class _Bar(_Foo):
    def __init__(self):
        super().__init__()
        self._b = self._a


class _MyGzipReader(_GzipReader):
    def __init__(self, fp):
        super().__init__(fp)
        self._my_last_mtime = self._last_mtime
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
```bash
❯ pyright example.py
/home/thughes/git/pgzip/example.py
  /home/thughes/git/pgzip/example.py:22:36 - error: Cannot access member "_last_mtime" for type "_MyGzipReader"
    Member "_last_mtime" is unknown (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
```

**Comment by erictraut** [comment]
The member `_last_mtime` isn't specified class `_GzipReader` as defined by [gzip.pyi](https://github.com/python/typeshed/blob/40c639fcb0e2c2628fff019e4b769fd364c46665/stdlib/gzip.pyi#L158). Consequently, this isn't considered an "access to a private member". It's considered an "access to a non-existent member" which always generates an error.

If you think that `_last_mtime` should be defined in the `_GzipReader` class, you can file a bug report (or a PR) in the [typeshed project](https://github.com/python/typeshed/issues). All of the python type checkers use the typeshed stubs as the source of truth about types in the python stdlib.

**Comment by bhearsum** [comment]
Is there a suggested workaround for this, aside from ignoring? There are many cases where it is useful to call non-public methods. For example, working around bugs or missing features in stdlib or 3rd party libraries. Ideally, being able to amend the types would be useful.

**Comment by erictraut** [comment]
I recommend against calling non-public methods if you're writing code that you intend to maintain over time. It makes the code fragile because these implementation details can easily change in the next release of the library.

> Is there a suggested workaround for this, aside from ignoring

If you find that a type definition in a public library or stub is incomplete, consider filing a bug report — or better yet, a PR that fixes the problem. When fixes like this are contributed back, everyone benefits. While you're waiting for your fix to be accepted, you can choose to either create your own project-local stub file that addresses the omission. Or you can use `# type: ignore` or `# pyright: ignore` to suppress the diagnostic in your code.
